evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -170,20 +171,32 @@ def _initialize_oversampler(self, X, y, sampler_class):
             y (pd.Series): Target features
             sampler_class (imblearn.BaseSampler): The sampler we want to initialize
         """
-        _, y_ww = self._prepare_data(X, y)
+        _, y_pd = self._prepare_data(X, y)
         sampler_params = {
             k: v
             for k, v in copy.copy(self.parameters).items()
-            if k not in ["sampling_ratio", "sampling_ratio_dict"]
+            if k not in ["sampling_ratio", "sampling_ratio_dict", "k_neighbors_default"]
         }
```

Comment:
Do we need the copying here?

```
sampler_params = {k: self.parameters[k] for k  ["sampling_ratio", "sampling_ratio_dict", "k_neighbors_default"]}
```

Reply:
I think we technically don't, I had just added it to be safer. I can remove it though!